s3/s3.go

```diff
@@ -0,0 +1,191 @@
+package s3
+
+import (
+	"bytes"
+	"context"
+	"errors"
+	"fmt"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/aws/retry"
+	awsconfig "github.com/aws/aws-sdk-go-v2/config"
+	"github.com/aws/aws-sdk-go-v2/credentials"
+	"github.com/aws/aws-sdk-go-v2/feature/s3/manager"
+	"github.com/aws/aws-sdk-go-v2/service/s3"
+	"github.com/aws/aws-sdk-go-v2/service/s3/types"
+)
+
+// Storage interface that is implemented by storage providers
+type Storage struct {
+	svc            *s3.Client
+	downloader     *manager.Downloader
+	uploader       *manager.Uploader
+	requestTimeout time.Duration
+	bucket         string
+}
+
+// New creates a new storage
+func New(config ...Config) *Storage {
+	// Set default config
+	cfg := configDefault(config...)
+
+	// Create s3 session
+	// If config fields of credentials not given, credentials are using from the environment variables (AWS_ACCESS_KEY_ID, AWS_SECRET_ACCESS_KEY), ~/.aws/credentials, or EC2 instance role.
```

Comment:
We should put this in the readme. Or better yet just link to https://aws.github.io/aws-sdk-go-v2/docs/configuring-sdk/#specifying-credentials